src/Vipps.net/Infrastructure/VippsHttpClient.cs

```diff
@@ -3,24 +3,25 @@
 using System.Net.Http;
 using System.Threading;
 using System.Threading.Tasks;
+using Vipps.net.Helpers;
 
 namespace Vipps.net.Infrastructure
 {
     public class VippsHttpClient : IVippsHttpClient
     {
         private HttpClient _httpClient;
         private readonly TimeSpan DefaultTimeOut = TimeSpan.FromSeconds(100);
+        private readonly VippsConfigurationOptions _options;
 
-        public VippsHttpClient() { }
-
-        public VippsHttpClient(HttpClient httpClient)
+        public VippsHttpClient(HttpClient httpClient, VippsConfigurationOptions options)
         {
             _httpClient = httpClient;
+            _options = options;
         }
 
         public Uri BaseAddress
         {
-            get { return HttpClient.BaseAddress; }
+            get { return new Uri(UrlHelper.GetBaseUrl(_options.UseTestMode)); }
```

Comment:
This should return the actual HttpClients base address, as a custom HttpClient with a non-default base address can be provided in the constructor.